server/submissionWorkflow/api.ts

```diff
@@ -0,0 +1,66 @@
+import app, { wrap } from 'server/server';
+import { ForbiddenError } from 'server/utils/errors';
+
+import { getPermissions } from './permissions';
+import {
+	createSubmissionWorkflow,
+	updateSubmissionWorkflow,
+	destroySubmissionWorkFlow,
+} from './queries';
+
+const getRequestIds = (req) => {
+	const user = req.user || {};
+	return {
+		userId: user.id,
+		communityId: req.body.communityId,
+		collectionId: req.body.id || null,
+	};
+};
+
+app.post(
+	'/api/submissionWorkflows',
+	wrap(async (req, res) => {
+		const requestIds = getRequestIds(req);
+		const permissions = await getPermissions(requestIds);
+		if (!permissions.create) {
+			throw new ForbiddenError();
+		}
+		const workflow = await createSubmissionWorkflow(req.body);
+		return res.status(201).json(workflow);
+	}),
+);
+
+app.put(
+	'/api/submissionWorkflows',
+	wrap(async (req, res) => {
+		const requestIds = getRequestIds(req);
+		const permissions = await getPermissions(requestIds);
+		if (!permissions.update) {
+			throw new ForbiddenError();
+		}
+		const updatedValues = await updateSubmissionWorkflow(
+			{
+				...req.body,
+				submissionWorkflowId: req.body.id,
```

Comment:
4️⃣ Consider that this value was never checked by `getPermissions`. What would happen in an attacker provided the ID of a `Collection` they own along with the ID of a `SubmissionWorkflow` they don't own? Your code should protect against this — I'd suggest by using `collectionId` instead of `id` in your `SubmissionWorkflow.update()` query. This is a good case to consider when writing API unit tests, too.